### ISSUE TITLE
qgis3 bottle filename incorrect

### ISSUE THREAD

**Issue by jonathanmccormack** [OSGeo/homebrew-osgeo4mac#432]
Looks like travis [successfully built](https://travis-ci.com/OSGeo/homebrew-osgeo4mac/jobs/138647717) the bottle for `qgis3` but homebrew cannot find the bottle on bintray:

```
$ brew install qgis3
==> Installing qgis3 from osgeo/osgeo4mac
==> Downloading https://dl.bintray.com/homebrew-osgeo/osgeo-bottles/qgis3-3.2.1.high_sierra.bottle.tar.gz

curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "qgis3"
Download failed: https://dl.bintray.com/homebrew-osgeo/osgeo-bottles/qgis3-3.2.1.high_sierra.bottle.tar.gz
Warning: Bottle installation failed: building from source.
```

A quick look shows that the bottle was deployed as `qgis3--3.2.1.high_sierra.bottle.tar.gz` but the JSON file is pointing to the filename `qgis3-3.2.1.high_sierra.bottle.tar.gz`:

```
[Bintray Upload] Reading descriptor file: travis/bintray-upload.json
[Bintray Upload] Uploading file 'bottles/qgis3--3.2.1.high_sierra.bottle.json' to qgis3--3.2.1.high_sierra.bottle.json
[Bintray Upload] Bintray response: 201 Created. success
[Bintray Upload] Uploading file 'bottles/qgis3--3.2.1.high_sierra.bottle.tar.gz' to qgis3--3.2.1.high_sierra.bottle.tar.gz
[Bintray Upload] Bintray response: 201 Created. success
[Bintray Upload] Uploading file 'bottles/qgis3--3.2.1.sierra.bottle.json' to qgis3--3.2.1.sierra.bottle.json
[Bintray Upload] Bintray response: 201 Created. success
[Bintray Upload] Uploading file 'bottles/qgis3--3.2.1.sierra.bottle.tar.gz' to qgis3--3.2.1.sierra.bottle.tar.gz
[Bintray Upload] Bintray response: 201 Created. success
[Bintray Upload] Publishing version '0.1' of package 'bottles'...
[Bintray Upload] Bintray response: 200 OK.
```

**Comment by nickrobison** [comment]
@jonathanmccormack Thanks for the heads up. I'm not sure why the Homebrew bottling process inserted the double hyphens, but I've re-uploaded the bottles and they should be working now.

Give it a try and let me know if you run into any other issues. Thanks!

**Comment by fjperini** [comment]
@nickrobison Solved!

**Comment by jonathanmccormack** [comment]
Thanks, that did the trick. It looks like something in the config / bottling builds the filename including an extra hyphen (and it looks like this impacts more than just qgis3):
```
$ brew install qgis3
==> Installing qgis3 from osgeo/osgeo4mac
==> Downloading https://dl.bintray.com/homebrew-osgeo/osgeo-bottles/qgis3-3.2.1.high_sierra.bottle.tar.gz
######################################################################## 100.0%
==> Pouring qgis3--3.2.1.high_sierra.bottle.tar.gz
```
 Might be related to Homebrew core [issue #4624](https://github.com/Homebrew/brew/issues/4624), which references a new "--" format.

**Comment by nickrobison** [comment]
@jonathanmccormack Thanks for the extra info! Curious as to why hombre is bottling with the double hypen, but downloading as the single hyphen and then renaming to the double hyphen. I guess we'll just have to monitor it and see how things progress from here.

**Comment by fjperini** [comment]
@nickrobison You will have to rename the bottle again and review what is happening.

**Comment by jonathanmccormack** [comment]
Not sure what the proper solution for the osgeo4mac travis scripts, but for reference maybe take a look at the discussion on Homebrew core [PR #4612](https://github.com/Homebrew/brew/pull/4612#commitcomment-29995084) and add something like `mv "$file" "$(echo "$file" | sed s/--/-/)"` before uploading to bintray.

**Comment by nickrobison** [comment]
Ok. I’ll make the changes tomorrow. Sorry for the hassle.

**Comment by nickrobison** [comment]
@jonathanmccormack @fjperini I believe I've renamed the bottles and addressed the core issue with b336104.

Please let me know if you're still having issues, I may have gotten the names mixed up a bit!

**Comment by fjperini** [comment]
@nickrobison @luisspuerto I'll do a test, I'll upload a change I made to use `ccache`. Tested on my machine, it works correctly!
